misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt

```diff
@@ -10,17 +10,39 @@ import javax.inject.Inject
 class FakeRedis : Redis {
   @Inject lateinit var clock: FakeClock
 
-  // The value type stored in our key-value store
+  // The value type stored in our key-data store. Data is always stored as ByteArrays.
+  // We manually define an equals and hashcode function as kotlin uses referential equality when comparing arrays,
+  // whereas we care about structural equality.
   private data class Value (
-    val value: String,
+    val data: ByteArray,
```

Comment:
should we just use okio `ByteString` instead? not a fan of the odd equality nature of ByteArray